### PR TITLE
Add Tools->Basic Fixup

### PR DESCRIPTION
**Pull request by windymilla**
Slightly different to GG1, this displays all the "errors", and user Ctrl/Cmd-clicks error to "process" it, i.e. fix it. In GG1, it just silently makes the changes (not necessarily desirable).

Future boosts to the checker dialog (see issue #193) may make this tool better.

Another possible future improvement would be a button to say "Make all suggested changes" after user has reviewed and removed (via right-click) any dodgy ones).

### CODE REVIEW

File: src/guiguts/misc_tools.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+
+logger = logging.getLogger(__package__)
+
+BLOCK_TYPES = "[$*XxFf]"
+POEM_TYPES = "[Pp]"
+
+
+def process_fixup(checker_entry: CheckerEntry) -> None:
+    """Process the fixup error."""
+    if checker_entry.text_range:
```
Comment (cpeel) [inline src/guiguts/misc_tools.py:18]: If I've put an entire function within an if I like to negate the condition and return, then fall through to the actual code so I don't have to indent the whole function. Clearly just a personal preference and what you have works great, but I noticed it and thought I'd mention it.

**Review by cpeel — APPROVED** [review]

File: src/guiguts/misc_tools.py (hunk 1 above)
Comment (windymilla) [reply src/guiguts/misc_tools.py]: Oh yes, thanks! I found a couple other small things while fixing that too!

**Comment by windymilla** [comment]
Testing notes for @srjfoo 
Should be suitable for running on a CTF.
Checks things like trailing spaces, `multiple      spaces` between words, spaces before punctuation, etc.
Doesn't check lines inside block markup `/*`, `/$`, etc
Ignore poetry line numbers (inside `/P` markup) as far as spacing is concerned, e.g.
```
/P
    Here is the first line of my poem        1
P/
It covers most of what the GG1 version checks and fixes

**Comment by srjfoo** [comment]
Not sure whether or not these need to be fixed, but the two things that jumped out at me were that it would be nice to

1. Omit proofers' notes from basic fixup, as the no-wrap is omitted. Apparently a some proofers like to put multiple sequential spaces inside their notes.
2. It would be good to have some way of indicating "I've seen this before, it's okay, please don't show it to me again." One of my test projects had about 20 instances of `whilst`, which basic fixup identifed as a possible `1/l` error. (Interestingly, the same word showed up once in another book, and was identified as a scanno.)

On macOS, `cmd`+left click "fixed" the error. `ctrl`+left-click, right-click on a mouse, and two-finger-right-click-alias all removed the line from the list.

Are all of these expected behavior?

**Comment by windymilla** [comment]
> Not sure whether or not these need to be fixed, but the two things that jumped out at me were that it would be nice to
> 
> 1. Omit proofers' notes from basic fixup, as the no-wrap is omitted. Apparently a some proofers like to put multiple sequential spaces inside their notes.
> 2. It would be good to have some way of indicating "I've seen this before, it's okay, please don't show it to me again." One of my test projects had about 20 instances of `whilst`, which basic fixup identifed as a possible `1/l` error. (Interestingly, the same word showed up once in another book, and was identified as a scanno.)
> 
> On macOS, `cmd`+left click "fixed" the error. `ctrl`+left-click, right-click on a mouse, and two-finger-right-click-alias all removed the line from the list.
> 
> Are all of these expected behavior?

All of that is expected. I've added a bit more to issue #193, which should deal with point 2.

I'm not going to do anything about point 1 for now.

**Comment by windymilla** [comment]
Latest commit should stop if reporting `whilst` - good catch @srjfoo

**Review by srjfoo — APPROVED** [review]

**Pull request merged**